hs_composite_resource/models.py

```diff
@@ -37,6 +44,81 @@ def set_default_logical_file(self):
+        return None
+
+    def get_folder_aggregation_type_to_set(self, dir_path):
+        """Returns an aggregation (file type) type that the specified folder *dir_path* can
+        possibly be set to.
+
+        :param dir_path: Resource file directory path (full folder path starting with resource id)
+        for which the possible aggregation type that can be set needs to be determined
+
+        :return If the specified folder is already represents an aggregation or does
+        not contain suitable file(s) then returns "" (empty string). If the specified folder
+        contains only the files that meet the requirements of a supported aggregation, and
+        does not contain other folders or does not have a parent folder then return the
+        class name of that matching aggregation type.
+        """
+        aggregation_type_to_set = ""
+        if self.get_folder_aggregation_object(dir_path) is not None:
+            # target folder is already an aggregation
+            return aggregation_type_to_set
+
+        istorage = self.get_irods_storage()
+        store = istorage.listdir(dir_path)
+        if store[0]:
+            # seems there are folders under dir_path - no aggregation type can be set if the target
+            # folder contains other folders
+            return aggregation_type_to_set
+
+        files_in_folder = [res_file for res_file in self.files.all()
+                           if res_file.dir_path == dir_path]
+        if not files_in_folder:
+            # folder is empty
+            return aggregation_type_to_set
+        if len(files_in_folder) > 1:
+            # check for geo feature
+            aggregation_type_to_set = GeoFeatureLogicalFile.check_files_for_aggregation_type(
+                files_in_folder)
+            if aggregation_type_to_set:
+                return aggregation_type_to_set
+
+            # check for raster
+            aggregation_type_to_set = GeoRasterLogicalFile.check_files_for_aggregation_type(
+                files_in_folder)
+            if aggregation_type_to_set:
+                return aggregation_type_to_set
+        else:
+            # check for raster
+            aggregation_type_to_set = GeoRasterLogicalFile.check_files_for_aggregation_type(
+                files_in_folder)
+            if aggregation_type_to_set:
+                return aggregation_type_to_set
+            # check for NetCDF aggregation type
+            aggregation_type_to_set = NetCDFLogicalFile.check_files_for_aggregation_type(
+                files_in_folder)
+            if aggregation_type_to_set:
+                return aggregation_type_to_set
+            # check for TimeSeries aggregation type
+            aggregation_type_to_set = TimeSeriesLogicalFile.check_files_for_aggregation_type(
+                files_in_folder)
+            if aggregation_type_to_set:
+                return aggregation_type_to_set
+
+        return aggregation_type_to_set
```

Comment:
It'd be easier to read if all the cases where an empty string is returned, you just return an empty string.  That way, there is no question that these cases are returning an empty string.

I'm curious why you've decided to use an empty string instead of None.  None is a singleton, which can slightly improve performance over creating an empty string.